// rtl-css-js/dist/cjs/core.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var convert = require('./convert-51dab0c6.js');



exports.arrayToObject = convert.arrayToObject;
exports.calculateNewBackgroundPosition = convert.calculateNewBackgroundPosition;
exports.calculateNewTranslate = convert.flipTransformSign;
exports.canConvertValue = convert.canConvertValue;
exports.convert = convert.convert;
exports.convertProperty = convert.convertProperty;
exports.flipSign = convert.flipSign;
exports.flipTransformSign = convert.flipTransformSign;
exports.getPropertyDoppelganger = convert.getPropertyDoppelganger;
exports.getValueDoppelganger = convert.getValueDoppelganger;
exports.getValuesAsList = convert.getValuesAsList;
exports.handleQuartetValues = convert.handleQuartetValues;
exports.includes = convert.includes;
exports.isBoolean = convert.isBoolean;
exports.isFunction = convert.isFunction;
exports.isNullOrUndefined = convert.isNullOrUndefined;
exports.isNumber = convert.isNumber;
exports.isObject = convert.isObject;
exports.isString = convert.isString;
exports.propertiesToConvert = convert.propertiesToConvert;
exports.propertyValueConverters = convert.propertyValueConverters;
exports.propsToIgnore = convert.propsToIgnore;
exports.splitShadow = convert.splitShadow;
exports.valuesToConvert = convert.valuesToConvert;
