// @chakra-ui/utils/dist/index.mjs
import "./chunk-WBQAMGXK.mjs";
import {
  detectBrowser,
  detectDeviceType,
  detectOS,
  detectTouch
} from "./chunk-DGFLLHTM.mjs";
import {
  walkObject
} from "./chunk-DVFODTG7.mjs";
import {
  determineLazyBehavior
} from "./chunk-XHETS734.mjs";
import {
  clampValue,
  countDecimalPlaces,
  maxSafeInteger,
  minSafeInteger,
  percentToValue,
  roundValueToStep,
  toPrecision,
  valueToPercent
} from "./chunk-YAFHXCR4.mjs";
import {
  PanSession
} from "./chunk-SV3JYFGC.mjs";
import {
  addPointerEvent,
  extractEventInfo,
  getPointerEventName,
  getViewportPointFromEvent,
  isMouseEvent,
  isMultiTouchEvent,
  isTouchEvent,
  wrapPointerEventHandler
} from "./chunk-LCE7F24A.mjs";
import {
  arrayToObjectNotation,
  breakpoints,
  isCustomBreakpoint,
  isResponsiveObjectLike,
  mapResponsive,
  objectToArrayNotation
} from "./chunk-FGAEJGLB.mjs";
import {
  analyzeBreakpoints,
  px,
  toMediaQueryString
} from "./chunk-FDX7JCYE.mjs";
import {
  default as default2,
  filterUndefined,
  fromEntries,
  get,
  getCSSVar,
  getWithDefault,
  memoize,
  memoizedGet,
  objectFilter,
  objectKeys,
  omit,
  pick,
  split
} from "./chunk-YTQ3XZ3T.mjs";
import {
  addItem,
  chunk,
  getFirstItem,
  getLastItem,
  getNextIndex,
  getNextItem,
  getNextItemFromSearch,
  getPrevIndex,
  getPrevItem,
  removeIndex,
  removeItem
} from "./chunk-YTAYUX3P.mjs";
import {
  closest,
  focusNextTabbable,
  focusPreviousTabbable,
  getAllFocusable,
  getAllTabbable,
  getFirstFocusable,
  getFirstTabbableIn,
  getLastTabbableIn,
  getNextTabbable,
  getPreviousTabbable
} from "./chunk-5LUSIWEA.mjs";
import {
  flatten
} from "./chunk-W5Y7LCVY.mjs";
import {
  focus
} from "./chunk-QKXRP2IX.mjs";
import {
  hasDisplayNone,
  hasFocusWithin,
  hasNegativeTabIndex,
  hasTabIndex,
  isActiveElement,
  isContentEditable,
  isDisabled,
  isFocusable,
  isHidden,
  isInputElement,
  isTabbable
} from "./chunk-P6S57EDQ.mjs";
import {
  addDomEvent,
  ariaAttr,
  canUseDOM,
  contains,
  cx,
  dataAttr,
  getActiveElement,
  getEventWindow,
  getOwnerDocument,
  getOwnerWindow,
  getRelatedTarget,
  isBrowser,
  isElement,
  isHTMLElement,
  isRightClick,
  normalizeEventKey
} from "./chunk-O3SWHQEE.mjs";
import {
  callAll,
  callAllHandlers,
  compose,
  distance,
  error,
  noop,
  once,
  pipe,
  runIfFn,
  warn
} from "./chunk-M3TFMUOL.mjs";
import {
  __DEV__,
  __TEST__,
  isArray,
  isCssVar,
  isDefined,
  isEmpty,
  isEmptyArray,
  isEmptyObject,
  isFunction,
  isInputEvent,
  isNotEmptyObject,
  isNotNumber,
  isNull,
  isNumber,
  isNumeric,
  isObject,
  isRefObject,
  isString,
  isUndefined
} from "./chunk-Y5FGD7DM.mjs";
import "./chunk-NHABU752.mjs";

// src/index.ts
export * from "css-box-model";
export {
  PanSession,
  __DEV__,
  __TEST__,
  addDomEvent,
  addItem,
  addPointerEvent,
  analyzeBreakpoints,
  ariaAttr,
  arrayToObjectNotation,
  breakpoints,
  callAll,
  callAllHandlers,
  canUseDOM,
  chunk,
  clampValue,
  closest,
  compose,
  contains,
  countDecimalPlaces,
  cx,
  dataAttr,
  detectBrowser,
  detectDeviceType,
  detectOS,
  detectTouch,
  determineLazyBehavior,
  distance,
  error,
  extractEventInfo,
  filterUndefined,
  flatten,
  focus,
  focusNextTabbable,
  focusPreviousTabbable,
  fromEntries,
  get,
  getActiveElement,
  getAllFocusable,
  getAllTabbable,
  getCSSVar,
  getEventWindow,
  getFirstFocusable,
  getFirstItem,
  getFirstTabbableIn,
  getLastItem,
  getLastTabbableIn,
  getNextIndex,
  getNextItem,
  getNextItemFromSearch,
  getNextTabbable,
  getOwnerDocument,
  getOwnerWindow,
  getPointerEventName,
  getPrevIndex,
  getPrevItem,
  getPreviousTabbable,
  getRelatedTarget,
  getViewportPointFromEvent,
  getWithDefault,
  hasDisplayNone,
  hasFocusWithin,
  hasNegativeTabIndex,
  hasTabIndex,
  isActiveElement,
  isArray,
  isBrowser,
  isContentEditable,
  isCssVar,
  isCustomBreakpoint,
  isDefined,
  isDisabled,
  isElement,
  isEmpty,
  isEmptyArray,
  isEmptyObject,
  isFocusable,
  isFunction,
  isHTMLElement,
  isHidden,
  isInputElement,
  isInputEvent,
  isMouseEvent,
  isMultiTouchEvent,
  isNotEmptyObject,
  isNotNumber,
  isNull,
  isNumber,
  isNumeric,
  isObject,
  isRefObject,
  isResponsiveObjectLike,
  isRightClick,
  isString,
  isTabbable,
  isTouchEvent,
  isUndefined,
  mapResponsive,
  maxSafeInteger,
  memoize,
  memoizedGet,
  default2 as mergeWith,
  minSafeInteger,
  noop,
  normalizeEventKey,
  objectFilter,
  objectKeys,
  objectToArrayNotation,
  omit,
  once,
  percentToValue,
  pick,
  pipe,
  px,
  removeIndex,
  removeItem,
  roundValueToStep,
  runIfFn,
  split,
  toMediaQueryString,
  toPrecision,
  valueToPercent,
  walkObject,
  warn,
  wrapPointerEventHandler
};
