// superagentai-js/Client.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuperAgentClient = void 0;
const Client_1 = require("./api/resources/agent/client/Client");
const Client_2 = require("./api/resources/agentDocuments/client/Client");
const Client_3 = require("./api/resources/tags/client/Client");
const Client_4 = require("./api/resources/agentTools/client/Client");
const Client_5 = require("./api/resources/auth/client/Client");
const Client_6 = require("./api/resources/user/client/Client");
const Client_7 = require("./api/resources/apiToken/client/Client");
const Client_8 = require("./api/resources/documents/client/Client");
const Client_9 = require("./api/resources/prompts/client/Client");
const Client_10 = require("./api/resources/tools/client/Client");
const Client_11 = require("./api/resources/traces/client/Client");
class SuperAgentClient {
    constructor(_options) {
        this._options = _options;
    }
    get agent() {
        var _a;
        return ((_a = this._agent) !== null && _a !== void 0 ? _a : (this._agent = new Client_1.Agent(this._options)));
    }
    get agentDocuments() {
        var _a;
        return ((_a = this._agentDocuments) !== null && _a !== void 0 ? _a : (this._agentDocuments = new Client_2.AgentDocuments(this._options)));
    }
    get tags() {
        var _a;
        return ((_a = this._tags) !== null && _a !== void 0 ? _a : (this._tags = new Client_3.Tags(this._options)));
    }
    get agentTools() {
        var _a;
        return ((_a = this._agentTools) !== null && _a !== void 0 ? _a : (this._agentTools = new Client_4.AgentTools(this._options)));
    }
    get auth() {
        var _a;
        return ((_a = this._auth) !== null && _a !== void 0 ? _a : (this._auth = new Client_5.Auth(this._options)));
    }
    get user() {
        var _a;
        return ((_a = this._user) !== null && _a !== void 0 ? _a : (this._user = new Client_6.User(this._options)));
    }
    get apiToken() {
        var _a;
        return ((_a = this._apiToken) !== null && _a !== void 0 ? _a : (this._apiToken = new Client_7.ApiToken(this._options)));
    }
    get documents() {
        var _a;
        return ((_a = this._documents) !== null && _a !== void 0 ? _a : (this._documents = new Client_8.Documents(this._options)));
    }
    get prompts() {
        var _a;
        return ((_a = this._prompts) !== null && _a !== void 0 ? _a : (this._prompts = new Client_9.Prompts(this._options)));
    }
    get tools() {
        var _a;
        return ((_a = this._tools) !== null && _a !== void 0 ? _a : (this._tools = new Client_10.Tools(this._options)));
    }
    get traces() {
        var _a;
        return ((_a = this._traces) !== null && _a !== void 0 ? _a : (this._traces = new Client_11.Traces(this._options)));
    }
}
exports.SuperAgentClient = SuperAgentClient;
