// @chakra-ui/theme/dist/chunk-BVEAU2MF.mjs
'use client'
import {
  stepperTheme
} from "./chunk-QHVXS45H.mjs";
import {
  switchTheme
} from "./chunk-XRIBTYHG.mjs";
import {
  tableTheme
} from "./chunk-FI633VWD.mjs";
import {
  tabsTheme
} from "./chunk-T6EHV47V.mjs";
import {
  tagTheme
} from "./chunk-VM3745AK.mjs";
import {
  textareaTheme
} from "./chunk-TN6RO673.mjs";
import {
  tooltipTheme
} from "./chunk-J54NBX2J.mjs";
import {
  progressTheme
} from "./chunk-BA7PF5LB.mjs";
import {
  radioTheme
} from "./chunk-BJIKSU3K.mjs";
import {
  selectTheme
} from "./chunk-ILLVD2I4.mjs";
import {
  skeletonTheme
} from "./chunk-T6E7UBRI.mjs";
import {
  skipLinkTheme
} from "./chunk-URZN3QPP.mjs";
import {
  sliderTheme
} from "./chunk-PUPKOASV.mjs";
import {
  spinnerTheme
} from "./chunk-HOBQJOCR.mjs";
import {
  statTheme
} from "./chunk-66QM3UU6.mjs";
import {
  kbdTheme
} from "./chunk-73PQYQVW.mjs";
import {
  linkTheme
} from "./chunk-NTN663XT.mjs";
import {
  listTheme
} from "./chunk-RRJIKBCH.mjs";
import {
  menuTheme
} from "./chunk-3XV5CZGX.mjs";
import {
  modalTheme
} from "./chunk-JIVTZ35T.mjs";
import {
  numberInputTheme
} from "./chunk-VEZNC4IQ.mjs";
import {
  pinInputTheme
} from "./chunk-HFED6QOE.mjs";
import {
  popoverTheme
} from "./chunk-F25MPA3T.mjs";
import {
  drawerTheme
} from "./chunk-5FAATIOO.mjs";
import {
  editableTheme
} from "./chunk-VJU277AG.mjs";
import {
  formTheme
} from "./chunk-OQ4UESHD.mjs";
import {
  formErrorTheme
} from "./chunk-66WVPDP6.mjs";
import {
  formLabelTheme
} from "./chunk-2RLCRBJQ.mjs";
import {
  headingTheme
} from "./chunk-SZS2B4EE.mjs";
import {
  inputTheme
} from "./chunk-FIZIJHUN.mjs";
import {
  breadcrumbTheme
} from "./chunk-CHG7JP5U.mjs";
import {
  buttonTheme
} from "./chunk-MMPN6JKT.mjs";
import {
  cardTheme
} from "./chunk-N5CTW2M4.mjs";
import {
  checkboxTheme
} from "./chunk-Y7UXPTSV.mjs";
import {
  closeButtonTheme
} from "./chunk-FC2RT2FC.mjs";
import {
  codeTheme
} from "./chunk-I7MXUFNA.mjs";
import {
  containerTheme
} from "./chunk-R5JELUQ3.mjs";
import {
  dividerTheme
} from "./chunk-6PVQMBW4.mjs";
import {
  accordionTheme
} from "./chunk-MTSOOEOV.mjs";
import {
  alertTheme
} from "./chunk-CJYJRQES.mjs";
import {
  avatarTheme
} from "./chunk-EYWUA2TC.mjs";
import {
  badgeTheme
} from "./chunk-4FUZMKSO.mjs";

// src/components/index.ts
var components = {
  Accordion: accordionTheme,
  Alert: alertTheme,
  Avatar: avatarTheme,
  Badge: badgeTheme,
  Breadcrumb: breadcrumbTheme,
  Button: buttonTheme,
  Checkbox: checkboxTheme,
  CloseButton: closeButtonTheme,
  Code: codeTheme,
  Container: containerTheme,
  Divider: dividerTheme,
  Drawer: drawerTheme,
  Editable: editableTheme,
  Form: formTheme,
  FormError: formErrorTheme,
  FormLabel: formLabelTheme,
  Heading: headingTheme,
  Input: inputTheme,
  Kbd: kbdTheme,
  Link: linkTheme,
  List: listTheme,
  Menu: menuTheme,
  Modal: modalTheme,
  NumberInput: numberInputTheme,
  PinInput: pinInputTheme,
  Popover: popoverTheme,
  Progress: progressTheme,
  Radio: radioTheme,
  Select: selectTheme,
  Skeleton: skeletonTheme,
  SkipLink: skipLinkTheme,
  Slider: sliderTheme,
  Spinner: spinnerTheme,
  Stat: statTheme,
  Switch: switchTheme,
  Table: tableTheme,
  Tabs: tabsTheme,
  Tag: tagTheme,
  Textarea: textareaTheme,
  Tooltip: tooltipTheme,
  Card: cardTheme,
  Stepper: stepperTheme
};

export {
  components
};
