// superagentai-js/serialization/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.apiToken = exports.auth = exports.agentTools = exports.agentDocuments = exports.tools = exports.prompts = exports.documents = exports.tags = exports.agent = void 0;
exports.agent = __importStar(require("./agent"));
exports.tags = __importStar(require("./tags"));
exports.documents = __importStar(require("./documents"));
exports.prompts = __importStar(require("./prompts"));
exports.tools = __importStar(require("./tools"));
__exportStar(require("./agent/client/requests"), exports);
exports.agentDocuments = __importStar(require("./agentDocuments"));
__exportStar(require("./agentDocuments/client/requests"), exports);
__exportStar(require("./tags/client/requests"), exports);
exports.agentTools = __importStar(require("./agentTools"));
__exportStar(require("./agentTools/client/requests"), exports);
exports.auth = __importStar(require("./auth"));
__exportStar(require("./auth/client/requests"), exports);
exports.apiToken = __importStar(require("./apiToken"));
__exportStar(require("./apiToken/client/requests"), exports);
__exportStar(require("./documents/client/requests"), exports);
__exportStar(require("./prompts/client/requests"), exports);
__exportStar(require("./tools/client/requests"), exports);
