// @chakra-ui/theme/dist/components/close-button.js
'use client'
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/close-button.ts
var close_button_exports = {};
__export(close_button_exports, {
  closeButtonTheme: () => closeButtonTheme
});
module.exports = __toCommonJS(close_button_exports);
var import_styled_system = require("@chakra-ui/styled-system");
var import_theme_tools = require("@chakra-ui/theme-tools");
var $size = (0, import_theme_tools.cssVar)("close-button-size");
var $bg = (0, import_theme_tools.cssVar)("close-button-bg");
var baseStyle = (0, import_styled_system.defineStyle)({
  w: [$size.reference],
  h: [$size.reference],
  borderRadius: "md",
  transitionProperty: "common",
  transitionDuration: "normal",
  _disabled: {
    opacity: 0.4,
    cursor: "not-allowed",
    boxShadow: "none"
  },
  _hover: {
    [$bg.variable]: "colors.blackAlpha.100",
    _dark: {
      [$bg.variable]: "colors.whiteAlpha.100"
    }
  },
  _active: {
    [$bg.variable]: "colors.blackAlpha.200",
    _dark: {
      [$bg.variable]: "colors.whiteAlpha.200"
    }
  },
  _focusVisible: {
    boxShadow: "outline"
  },
  bg: $bg.reference
});
var sizes = {
  lg: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.10",
    fontSize: "md"
  }),
  md: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.8",
    fontSize: "xs"
  }),
  sm: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.6",
    fontSize: "2xs"
  })
};
var closeButtonTheme = (0, import_styled_system.defineStyleConfig)({
  baseStyle,
  sizes,
  defaultProps: {
    size: "md"
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  closeButtonTheme
});
