// @chakra-ui/theme/dist/components/divider.js
'use client'
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/divider.ts
var divider_exports = {};
__export(divider_exports, {
  dividerTheme: () => dividerTheme
});
module.exports = __toCommonJS(divider_exports);
var import_styled_system = require("@chakra-ui/styled-system");
var baseStyle = (0, import_styled_system.defineStyle)({
  opacity: 0.6,
  borderColor: "inherit"
});
var variantSolid = (0, import_styled_system.defineStyle)({
  borderStyle: "solid"
});
var variantDashed = (0, import_styled_system.defineStyle)({
  borderStyle: "dashed"
});
var variants = {
  solid: variantSolid,
  dashed: variantDashed
};
var dividerTheme = (0, import_styled_system.defineStyleConfig)({
  baseStyle,
  variants,
  defaultProps: {
    variant: "solid"
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  dividerTheme
});
