// @chakra-ui/theme/dist/components/spinner.js
'use client'
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/spinner.ts
var spinner_exports = {};
__export(spinner_exports, {
  spinnerTheme: () => spinnerTheme
});
module.exports = __toCommonJS(spinner_exports);
var import_styled_system = require("@chakra-ui/styled-system");
var import_theme_tools = require("@chakra-ui/theme-tools");
var $size = (0, import_theme_tools.cssVar)("spinner-size");
var baseStyle = (0, import_styled_system.defineStyle)({
  width: [$size.reference],
  height: [$size.reference]
});
var sizes = {
  xs: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.3"
  }),
  sm: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.4"
  }),
  md: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.6"
  }),
  lg: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.8"
  }),
  xl: (0, import_styled_system.defineStyle)({
    [$size.variable]: "sizes.12"
  })
};
var spinnerTheme = (0, import_styled_system.defineStyleConfig)({
  baseStyle,
  sizes,
  defaultProps: {
    size: "md"
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  spinnerTheme
});
